shuup/core/tasks.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2020, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+import json
+
+from shuup.utils.importing import cached_load, load
+
+
+class TaskNotSerializableError(Exception):
+    """
+    Raised when the task can't be serialized.
+    """
+    pass
+
+
+class Task(object):
+    function = None     # type: str
+    kwargs = None       # type: dict
```

Comment:
Since the values can be `None`, the correct type comments here would be:

```
function = None     # type: Optional[str]
kwargs = None       # type: Optional[dict]
```

Or more preferably the `dict` type would be specified a bit more:

```
kwargs = None  # type: Optional[Dict[str, Any]]
```

Reply:
kwargs should be any dict that can be parsed into JSON, so maybe `Any` there is weird.

Reply:
Type hinting JSON is a long standing issue, but at least I always use the same type alias as Guido does https://github.com/python/typing/issues/182#issuecomment-199532520.